notebooks/working/5.0-mab-simulation-causal-graph.md

```diff
@@ -0,0 +1,994 @@
+
+#     alt_varying_variables = alt_var_vars
+#     # Specifying the column heading for the alternative id column in the
+#     # long format dataset
+#     custom_alt_id = "alternative_id"
+#     obs_id_column = "custom_id"
+#     sample_data[obs_id_column] = np.arange(sample_data.shape[0],
+#                                            dtype=int) + 1
+
+#     # Create a variable recording the choice column
+
+#     long_data = pl.convert_wide_to_long(wide_data=sample_data,
+#                                         ind_vars=ind_variables,
+#                                         alt_specific_vars=alt_varying_variables,
+#                                         availability_vars=availability_variables,
+#                                         obs_id_col=obs_id_column,
+#                                         choice_col=choice_var,
+#                                         new_alt_id_name=custom_alt_id)
+    sim_choice_var = 'sim_choice'
+    # Functions to generate the probabilities for each alternative
+
+    long_data = data
+
+    design_matrix, names = create_design_matrix(df_long=long_data,
+                                                specification_dict=mnl_specification,
+                                                names_dict=mnl_names,
+                                                alternative_id_col=alt_id_col)
+    mapping_matrix = create_mapping_matrix(df_long=long_data,
+                                           observation_id_col=obs_id_col)
+    probabilities = calculate_probabilities(betas=initial_betas,
+                                            design_matrix=design_matrix,
+                                            mapping_matrix=mapping_matrix)
+
+    data['probabilities'] = probabilities
+    data['cum_sum'] = 0
+    data['sim_choice'] = 0
+
+    for observation in data['observation_id'].unique():
+        probs_sum = data[data.observation_id==observation]['probabilities'].cumsum()
+        data.loc[data['observation_id']==observation,'cum_sum'] = probs_sum
+
+    observation_id_list = list(data.observation_id.unique())
+    u_random = np.random.uniform(size = len(data['observation_id'].unique()))
+
+    for u,obs in zip(u_random,observation_id_list):
+        data_sample = data[data['observation_id']==obs]
+        sorted_list = sorted(list(data_sample['mode_id'].unique()))
+        choices = dict.fromkeys(sorted_list, 0)
+        for alt in sorted_list:   
+            choices[alt] = np.where(u<=data_sample[data_sample['mode_id']==alt]['cum_sum'], 1, 0).item()
+            if choices[alt] == 1:
+                break
+        data.loc[data.observation_id==obs,'sim_choice'] = data['mode_id'].map(choices)
+    return data
+<!-- #endraw -->
+
+# Example Application Using Bike Data
+
+
+## Data Ingestion and Exploration
+
+```python
+# Create a variable for the path to the long format data for
+# the multinomial choice model
+PATH = '/Users/mobouzaghrane/Documents/GitHub/tr_b_causal_2020/'\
```

Comment:
Could you replace this with a relative path so that it'll work for other users?

Reply:
```suggestion
PATH = '../../'\
```

Reply:
Will do!